bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/ircc/IrccProtocol.java

```diff
@@ -0,0 +1,823 @@
+                    if (!StringUtils.equalsIgnoreCase(cmdToSend, cmd)) {
+                        logger.debug("Transformed {} with map file '{}' to {}", cmd, cmdMap, cmdToSend);
+                    }
+                }
+            } catch (final TransformationException e) {
+                logger.debug("Failed to transform {} using map file '{}', exception={} - ignoring error", cmd, cmdMap,
+                        e.getMessage());
+            }
+        }
+
+        try {
+            cmdToSend = URLDecoder.decode(cmdToSend, "UTF-8");
+        } catch (final UnsupportedEncodingException e) {
+            logger.debug("Failed to decode {}, exception={} - ignoring error", cmdToSend, e.getMessage());
+        }
+
+        final int idx = cmdToSend == null ? -1 : cmdToSend.indexOf(':');
+
+        String protocol = IrccRemoteCommand.IRCC;
+        if (cmdToSend != null && idx >= 0) {
+            protocol = cmdToSend.substring(0, idx);
+            cmdToSend = cmdToSend.substring(idx + 1);
+        }
+
+        if (cmdToSend == null || StringUtils.isEmpty(cmdToSend)) {
+            logger.debug("Command was empty - ignoring");
+        } else if (StringUtils.equalsIgnoreCase(IrccRemoteCommand.IRCC, protocol)) {
+            sendIrccCommand(cmdToSend);
+        } else if (StringUtils.equalsIgnoreCase(IrccRemoteCommand.URL, protocol)) {
+            final HttpResponse resp = transport.executeGet(cmdToSend);
+            if (resp.getHttpCode() == HttpStatus.OK_200) {
+                logger.trace("Send of command {} was successful", cmdToSend);
+            } else if (resp.getHttpCode() == HttpStatus.SERVICE_UNAVAILABLE_503) {
+                logger.debug("URL service is unavailable (power off?)");
+            } else {
+                logger.debug("Bad return code from {}: {}", IrccClient.SRV_ACTION_SENDIRCC, resp);
+            }
+        } else {
+            logger.debug("Unknown protocol found for the send command: {}", cmd);
+        }
+    }
+
+    /**
+     * Send an IRCC command to the device
+     *
+     * @param cmdToSend the non-null, non-empty IRCC command to send
+     */
+    private void sendIrccCommand(final String cmdToSend) {
+        Validate.notEmpty(cmdToSend, "cmdToSend cannot be empty");
+
+        final HttpResponse resp = irccClient.executeSoap(transport, cmdToSend);
+        if (resp.getHttpCode() == HttpStatus.OK_200) {
+            logger.trace("Sending of IRCC command {} was successful", cmdToSend);
+        } else if (resp.getHttpCode() == HttpStatus.SERVICE_UNAVAILABLE_503) {
+            logger.debug("IRCC service is unavailable (power off?)");
+        } else if (resp.getHttpCode() == HttpStatus.INTERNAL_SERVER_ERROR_500) {
+            logger.debug("IRCC service returned a 500 - probably an unknown command: {}", cmdToSend);
+        } else {
+            logger.error("Bad return code from {}: {}", IrccClient.SRV_ACTION_SENDIRCC, resp);
```

Comment:
How come a bad code from sendContentUrl is logged at WARN but this is logged at ERROR?

Reply:
Good catch - done